bin/update-copyright-year

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -euf -o pipefail
+
+# shellcheck source=functions
+source "${BASH_SOURCE%/*}/../sanctuary-scripts/functions"
+
+if [[ "${RECURSING-false}" == false ]] && [[ -x scripts/update-copyright-year ]] ; then
+  RECURSING=true scripts/update-copyright-year
+  exit 0
+fi
+
+declare -r creator="${1-Sanctuary}"
+declare -r license="${2-LICENSE}"
```

Comment:
What's the logic which determines a script using command line arguments over a value in `.config`? Incidentally, could you teach me about `${1-foo}`-syntax?

Reply:
> What's the logic which determines a script using command line arguments over a value in `.config`?

This script is inconsistent in this respect, but for good reasons (I believe). It seems to me the script most likely to be used in non-Sanctuary projects. It's thus important that the name of the licence file is configurable (`check-required-files` actually requires a file named __LICENSE__, so this is only relevant to projects using scripts à la carte).

I'd like to support those who wish to use *only* this script. It would be burdensome and a little messy to require a configuration file in order to use a single shell script.

Perhaps I'm overthinking this. It would certainly be more consistent to move these to __.config__, and to remove from `check-required-files` the hard-coded reference to __LICENSE__. Is this what you suggest?

> Incidentally, could you teach me about `${1-foo}`-syntax?

The first thing to know is that `$var` is shorthand for `${var}`:

```console
$ VAR=hello bash -c 'var=world ; echo "$VAR $var"'
hello world

$ VAR=hello bash -c 'var=world ; echo "${VAR} ${var}"'
hello world
```

As you're no doubt aware, `$1`, `$2`, `$3`, etc. refer to command-line arguments. We now know that these can be written as `${1}`, `${2}`, `${3}`, etc.

[§ 3.5.3 Shell Parameter Expansion][1]:

> `${parameter:-word}`
>
> If _parameter_ is unset or null, the expansion of _word_ is substituted. Otherwise, the value of _parameter_ is substituted.

```console
$ bash -c 'foo=42 ; bar= ; echo "foo: ${foo:-default}, bar: ${bar:-default}, baz: ${baz:-default}"'
foo: 42, bar: default, baz: default
```

We are omitting the colon, though, which changes the handling of null:

> When not performing substring expansion, using the form described below (e.g., ‘`:-`’), Bash tests for a parameter that is unset or null. Omitting the colon results in a test only for a parameter that is unset. Put another way, if the colon is included, the operator tests for both `parameter`’s existence and that its value is not null; if the colon is omitted, the operator tests only for existence.

```console
$ bash -c 'foo=42 ; bar= ; echo "foo: ${foo-default}, bar: ${bar-default}, baz: ${baz-default}"'
foo: 42, bar: , baz: default
```

So, `${1-Sanctuary}` evaluates to the value of the first command-line argument if at least one command-line argument is provided; `Sanctuary` otherwise. As is so often the case with Bash, this convenient functionality is hidden behind arcane syntax few will ever discover.


[1]: https://www.gnu.org/software/bash/manual/bashref.html#Shell-Parameter-Expansion